Automattic/wp-calypso
# Domains: Update domain suggestions cards layout

#### Changes proposed in this Pull Request

This PR updates the layout of domain suggestions cards in the domain registration search page. This corresponds to task 8 of pcYYhz-e4-p2, which is part of a redesign of domains pages.

The main updates in this PR are:

- Featured domain suggestions are always shown in a full-width row - previously, they could be shown as two side-by-side cards if the searched domain name was short enough
- The colored bars indicating the quality of the suggestion ("Best match" or "Best alternative") were removed. They were replaced by badges with "Recommended" and "Best alternative" texts
- Both featured domain suggestions have the "Selected" button colored as primary now
- The "Sale" badge was reskinned to have the same format as the "Recommended" and "Best alternative" ones

This is what the domain suggestions page looked like before:

<img width="1055" alt="Screen Shot 2021-08-02 at 20 00 14" src="https://user-images.githubusercontent.com/5324818/127934732-3c1cf44d-d9df-4259-b5f3-e6dad631a315.png">

And what it looks like with this update:

<img width="1055" alt="Screen Shot 2021-08-11 at 12 35 25" src="https://user-images.githubusercontent.com/5324818/129059585-e55515d5-3ced-4bd0-b089-2e45e5c04ff6.png">

Mobile view before:

<img width="417" alt="Screen Shot 2021-08-02 at 20 02 44" src="https://user-images.githubusercontent.com/5324818/127934845-5bf52beb-88b3-494f-a216-8a2e1fa37725.png">

Mobile view after:

<img width="429" alt="Screen Shot 2021-08-11 at 12 36 42" src="https://user-images.githubusercontent.com/5324818/129059616-983ab796-92ed-4779-9ce8-b6414f6d72ac.png">

#### Testing instructions



Use the live Calypso link and navigate to the domain name search page (Upgrades > Domains > Add a domain > To this site). Do a few domain searches with different names and TLDs and ensure everything looks alright. Please test both desktop and mobile views to ensure both are OK.

It would also be great if you could take a look at the signup flow (`<calypso live URL>/start`) and check that the domain search there looks OK too, because it uses the same components that were updated here. 🙇‍♂️ It was a little tricky to not change things in that page.
